csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -59,6 +59,103 @@ bool isExactParallelSharedMemAccess(TensorView* tv) {
   return true;
 }
 
+// Check for conditions where the predicate cannot be removed
```

Comment:
Could you move this refactor to a separate PR?

Reply:
movd to https://github.com/NVIDIA/Fuser/pull/3098